### PR TITLE
Demo automatic personal data audit logs in Booshop sample

### PR DESCRIPTION
**Pull request by mofterdinger**
- Annotate entity Author with personal data annotations
- audit log config changes for updates on orders.currency_code

### CODE REVIEW

File: srv/src/main/java/my/bookshop/handlers/AdminServiceAuditHandler.java [hunk 1]
```diff
@@ -1,178 +0,0 @@
...
-import cds.gen.adminservice.AdminService_;
-import cds.gen.adminservice.Orders;
-import cds.gen.adminservice.Orders_;
-
-/**
- * A custom handler that creates AuditLog messages.
- */
-@Component
-@ServiceName(AdminService_.CDS_NAME)
-class AdminServiceAuditHandler implements EventHandler {
```
[the remark this answers is not in the document]
Comment (mofterdinger) [reply srv/src/main/java/my/bookshop/handlers/AdminServiceAuditHandler.java:42]: No, I removed it the DataModification and DataAccess logging via custom handler. I can add a ConfigChange logging, but what kind of config is available in the bookshop?

File: srv/src/main/java/my/bookshop/handlers/AdminServiceAuditHandler.java [hunk 2]
```diff
@@ -50,52 +44,46 @@
...
-	}
-
 	@Before(event = { CqnService.EVENT_CREATE })
 	public void beforeCreateOrder(Stream<Orders> orders) {
 		orders.forEach(order -> {
-			DataModification dataModification = createDataModification(order, null, Action.CREATE);
-			if (dataModification != null) {
-				this.auditLog.logDataModification(Arrays.asList(dataModification));
+			ConfigChange cfgChange = createConfigChange(order, null);
+			if (cfgChange != null) {
```
Comment (BraunMatthias) [inline srv/src/main/java/my/bookshop/handlers/AdminServiceAuditHandler.java:51]: `cfgChange `will nevver be null, won't it?

**Review by BraunMatthias — APPROVED** [review]
ConfigChange is much better for public demo as it is supposed to be sent manually

File: srv/src/main/java/my/bookshop/handlers/AdminServiceAuditHandler.java (hunk 2 above)
Comment (mofterdinger) [reply srv/src/main/java/my/bookshop/handlers/AdminServiceAuditHandler.java:51]: right, will never be null, will remove check

**Review by BraunMatthias — APPROVED** [review]

**Review by davidhunglam — APPROVED** [review]

**Pull request merged**